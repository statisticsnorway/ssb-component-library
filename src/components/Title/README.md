Title
========

> A component handling header elements

### Usage

#### HTML

```html
<div class="ssb-title"><h1>Title</h1></div>

<div class="ssb-title"><h2>Title</h2></div>
```

#### React

```jsx harmony
<Title size="1">Title</Title>

<Title size="2" negative>Title</Title>
```

Available props:

| Name       | Type          | Description  |
| ---------- | ------------- | ----- |
| children | node | Child nodes |
| className | string | Optional container class |
| negative | bool | Toggles text color |
| size | 1, 2, 3, 4, 5 or 6 | Changes header element size |
