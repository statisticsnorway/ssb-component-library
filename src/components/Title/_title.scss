.ssb-title {
  color: $ssb-dark-6;
  margin-top: 0;

  &.negative {
    color: $ssb-white;
  }

  @at-root {

    h1#{&} {
      @include roboto-condenced;
      font-size: 56px;
      font-weight: bold;
      line-height: 1.4;
    }

    h2#{&} {
      @include roboto-condenced;
      font-size: 28px;
      font-weight: bold;
      line-height: 1.5;
    }

    h3#{&} {
      @include roboto-condenced;
      font-size: 24px;
      font-weight: bold;
      line-height: 1.6;
    }

    h4#{&} {
      @include roboto;
      font-size: 20px;
      font-weight: bold;
      line-height: 1.7;
    }

    h5#{&} {
      @include roboto;
      font-size: 18px;
      font-weight: bold;
      line-height: 1.7;
    }

    h6#{&} {
      @include roboto;
      font-size: 16px;
      font-weight: bold;
      line-height: 1.7;
    }
  }
}
