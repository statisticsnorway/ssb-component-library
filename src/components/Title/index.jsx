import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { roboto, robotoCondenced } from '../../style/mixins';
import { ssbDark6, ssbWhite } from '../../style/colors';

const Title = ({ children, className, negative, size }) => (
	<TitleWrapper
		className={className}
		negative={negative}
	>
		{size === 1 && <h1>{children}</h1>}
		{size === 2 && <h2>{children}</h2>}
		{size === 3 && <h3>{children}</h3>}
		{size === 4 && <h4>{children}</h4>}
		{size === 5 && <h5>{children}</h5>}
		{size === 6 && <h6>{children}</h6>}
	</TitleWrapper>
);

const TitleWrapper = styled.div`
	color: ${props => (props.negative ? ssbWhite : ssbDark6)};

	h1 {
		${robotoCondenced};
		font-size: 56px;
		font-weight: bold;
		line-height: 1.4;
	}

	h2 {
		${robotoCondenced};
		font-size: 28px;
		font-weight: bold;
		line-height: 1.5;
	}

	h3 {
		${robotoCondenced};
		font-size: 24px;
		font-weight: bold;
		line-height: 1.6;
	}

	h4 {
		${roboto};
		font-size: 20px;
		font-weight: bold;
		line-height: 1.7;
	}

	h5 {
		${roboto};
		font-size: 18px;
		font-weight: bold;
		line-height: 1.7;
	}

	h6 {
		${roboto};
		font-size: 16px;
		font-weight: bold;
		line-height: 1.7;
	}
`;

Title.defaultProps = {
	negative: false,
	size: 1,
};

Title.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	negative: PropTypes.bool,
	size: PropTypes.number,
};

export default Title;
