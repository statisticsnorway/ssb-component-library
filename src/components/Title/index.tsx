import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface TitleProps {
  children: ReactNode
  className?: string
  negative?: boolean
  size?: 1 | 2 | 3 | 4 | 5 | 6
}

const Title: React.FC<TitleProps> = ({ children, className, negative = false, size = 1 }) => {
  const classNameStr = classNames('ssb-title', className, negative && 'negative')
  switch (size) {
    case 1:
      return <h1 className={classNameStr}>{children}</h1>
    case 2:
      return <h2 className={classNameStr}>{children}</h2>
    case 3:
      return <h3 className={classNameStr}>{children}</h3>
    case 4:
      return <h4 className={classNameStr}>{children}</h4>
    case 5:
      return <h5 className={classNameStr}>{children}</h5>
    case 6:
      return <h6 className={classNameStr}>{children}</h6>
    default:
      return <h1 className={classNameStr}>{children}</h1>
  }
}

export default Title
