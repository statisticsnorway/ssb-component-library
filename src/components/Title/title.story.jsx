import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Title from './index';

storiesOf('Title', module).addDecorator(centered)
	.add('Default', () => (
		<div>
			<Title size={1}>This is a h1 title</Title>
			<Title size={2}>This is a h2 title</Title>
			<Title size={3}>This is a h3 title</Title>
			<Title size={4}>This is a h4 title</Title>
			<Title size={5}>This is a h5 title</Title>
			<Title size={6}>This is a h6 title</Title>
		</div>
	));
