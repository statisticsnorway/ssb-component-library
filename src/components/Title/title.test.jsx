import React from 'react';
import {shallow} from 'enzyme';
import Title from './index';

describe('Text component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Title>Title</Title>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Sets negative class', () => {
		const wrapper = shallow(<Title negative>Title</Title>);
		expect(wrapper.find('.ssb-title').hasClass('negative')).toEqual(true);
	});
	test('Sets custom class', () => {
		const wrapper = shallow(<Title className="some-class">Title</Title>);
		expect(wrapper.find('.ssb-title').hasClass('some-class')).toEqual(true);
	});
	test('Renders correct default size', () => {
		const wrapper = shallow(<Title>Title</Title>);
		expect(wrapper.find('h1').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={1}>Title</Title>);
		expect(wrapper.find('h1').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={2}>Title</Title>);
		expect(wrapper.find('h2').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={3}>Title</Title>);
		expect(wrapper.find('h3').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={4}>Title</Title>);
		expect(wrapper.find('h4').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={5}>Title</Title>);
		expect(wrapper.find('h5').exists()).toEqual(true);
	});
	test('Renders correct size', () => {
		const wrapper = shallow(<Title size={6}>Title</Title>);
		expect(wrapper.find('h6').exists()).toEqual(true);
	});
});
