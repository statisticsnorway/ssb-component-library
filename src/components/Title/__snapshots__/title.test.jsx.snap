// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Matches the snapshot 1`] = `
<h1
  className="ssb-title"
>
  Title
</h1>
`;
